clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsBuilderTest.java

```diff
@@ -728,35 +730,36 @@ public void shouldThrowIllegalStateExceptionOnAppendWhenAborted(Args args) {
     @ParameterizedTest
     @ArgumentsSource(MemoryRecordsBuilderArgumentsProvider.class)
     public void testBuffersDereferencedOnClose(Args args) {
```

Comment:
This test case fails frequently. It seems to me it tries to verify the reference to buffer gets released. Hence, maybe we can check the `appendStream` instead of calling GC?

Reply:
> This test case fails frequently.

Yeah... I just hit this test case
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16783/11/testReport/org.apache.kafka.common.record/MemoryRecordsBuilderTest/Build___JDK_21_and_Scala_2_13____testBuffersDereferencedOnClose_Args__args_magic_1__bufferOffset_15__compression_org_apache_kafka_common_compress_GzipCompression_1e_/